library/firewall_lib.py

```diff
@@ -343,6 +372,42 @@ def try_set_zone_of_interface(module, _zone, interface):
     return (False, False)
 
 
+# Check that all of the ipset entries are ipv4, ipv6, or mac addresses
+# if not, fail the module
+# if they are, return "ipv4", "ipv6", or "mac"
+def get_ipset_entries_type(ipset_entries, module):
+    addr_type = None
+    is_mixed_addr_types = False
+    for entry in ipset_entries:
+        if check_mac(entry):
+            if addr_type is None:
+                addr_type = "mac"
+            elif addr_type != "mac":
+                is_mixed_addr_types = True
+        else:
+            if HAS_IPADDRESS:
+                addr = ip_interface(entry)
```

Comment:
IIRC, if the address is invalid then this will throw a `ValueError` exception. Do you need to handle it here? Or maybe in the wrapper `ip_interface()`.